docs/user_guide/configuration.md

```diff
@@ -0,0 +1,399 @@
+
+## Configuration Layers
+
+The Gateway's configuration follows a layered approach:
+
+1. **Spring Boot Configuration**: Core Spring Boot settings (application.yml)
+2. **GeOrchestra Data Directory**: External configuration files in a central location
+3. **Environment Variables**: Runtime configuration through environment variables
+4. **Docker Configuration**: When deployed with Docker, additional settings via Docker Compose
+
+## Configuration Files
+
+The geOrchestra Gateway configuration is primarily managed through YAML files located in the `datadir/gateway/` directory:
+
+- `application.yaml` - Core application properties
+- `gateway.yaml` - Gateway services and routing configuration
+- `security.yaml` - Authentication and authorization settings
+- `routes.yaml` - Route definitions 
+- `roles-mappings.yaml` - Role mapping configuration
+- `logging.yaml` - Logging configuration
+
+## Data Directory
+
+The Gateway reads its configuration from the geOrchestra "data directory". The location of this directory is specified using the `georchestra.datadir` environment property.
+
+The additional property sources are loaded using Spring Boot's `spring.config.import` property, for example:
+
+```yaml
+spring.config.import: ${georchestra.datadir}/default.properties,${georchestra.datadir}/gateway/gateway.yaml
+```
+
+### Docker Setup
+
+When using Docker, the data directory is typically mounted as a volume:
+
+```yaml
+volumes:
+  - georchestra_datadir:/etc/georchestra
+```
+
+The environment variable is set to point to this location:
+
+```yaml
+environment:
+  - JAVA_TOOL_OPTIONS=-Dgeorchestra.datadir=/etc/georchestra
+```
+
+### Configuration File Purposes
+
+Each configuration file in the data directory serves a specific purpose:
+
+| File | Purpose |
+|------|---------|
+| `application.yaml` | Contains filters applicable to all transfers through the gateway, and general Spring settings |
+| `gateway.yaml` | Contains access rules for all services available through the gateway |
+| `roles-mappings.yaml` | Maps roles returned by authentication providers to standardized geOrchestra roles |
+| `routes.yaml` | Contains list of routes for redirecting to the correct service URL based on criteria |
+| `security.yaml` | Contains all settings about authentication, including OAuth2 and pre-authentication |
+
+## Core Settings
+
+### Basic Application Properties
+
+```yaml
+spring:
+  application:
+    name: gateway
+  main:
+    web-application-type: reactive
+
+server:
+  port: 8080
+  compression:
+    enabled: true
+```
+
+### Profiles
+
+Spring profiles can be activated to enable specific functionality:
+
+```yaml
+spring:
+  profiles:
+    active: default,json-logs
+```
+
+Common profiles include:
+- `docker` - Settings for Docker environments
+- `json-logs` - Enable JSON-formatted structured logging
+- `logging_debug` - Enable more detailed logging
+- `preauth` - Enable pre-authentication mode
```

Comment:
Not self-explanatory. Would be good to provide an href there (https://groldan.github.io/georchestra-gateway/user_guide/authentication/#pre-authentication-via-http-headers right ?)

Reply:
[gotcha](https://github.com/georchestra/georchestra-gateway/compare/d096bea6af1d0cb0c2279e213e0f3bcb0463ab87..521fa3e62c58dcabc0fbc567fa0315f68677a7b8)